### ISSUE TITLE
ISystemMediaTransportControlsInterop.GetForWindow causes STATUS_ACCESS_VIOLATION

### ISSUE THREAD

**Issue by Sinono3** [microsoft/windows-rs#924]
Hi. My crate Souvlaki, was working fine with windows-rs 0.8 before I migrated to version 0.13 ([you can check the relevant part of the source code](https://github.com/Sinono3/souvlaki/blob/2421e2a98df8eede66adafa42c11e8c9256f595c/src/platform/windows/mod.rs#L39-L51))

The problem occurs when I try to create a `SystemMediaTransportControls` struct using an interop class. When migrating to windows-rs 0.13 a few errors due to changes in function arguments popped up, so this was the version I came up with after refactoring.

```rust
use bindings::Media::SystemMediaTransportControls;
use bindings::Win32::Media::MediaTransport::ISystemMediaTransportControlsInterop;
use bindings::Win32::Foundation::HWND;

let window_handle: WindowsHandle; // A handle to an HWND

let interop: ISystemMediaTransportControlsInterop =
    windows::factory::<SystemMediaTransportControls, ISystemMediaTransportControlsInterop>()?;

let controls: SystemMediaTransportControls = unsafe {
    interop.GetForWindow(
        HWND(window_handle.hwnd as isize)
    ) // The crash happens on this call 
}?; 
```

Here is the output of cargo run when running the only example in the crate:
```
error: process didn't exit successfully: `target\debug\examples\example.exe` (exit code: 0xc0000005, STATUS_ACCESS_VIOLATION)
The terminal process "C:\Users\User\.cargo\bin\cargo.exe 'run', '--package', 'souvlaki', '--example', 'example'" terminated with exit code: 3221225477.
```

I'm sure it's not a problem with the window handle, since it used to work fine with pretty much the same syntax.

**Comment by kennykerr** [comment]
Nothing jumps out at me. Can you share a complete and minimal repro?

**Comment by Sinono3** [comment]
Here's a repro with winit.
[repro.zip](https://github.com/microsoft/windows-rs/files/6742648/repro.zip)

**Comment by riverar** [comment]
ISystemMediaTransportControlsInterop inherits IInspectable but we're generating the below. The offsets are wrong so we end up calling into `IInspectable::GetIids` oops.

Generated:
```
...
pub struct ISystemMediaTransportControlsInterop(::windows::IUnknown);
impl ISystemMediaTransportControlsInterop {
    pub unsafe fn GetForWindow<'a, T: ::windows::Interface>(
        &self,
        appwindow: impl ::windows::IntoParam<'a, super::super::Foundation::HWND>,
    ) -> ::windows::Result<T> {
        let mut result__ = ::std::option::Option::None;
        (::windows::Interface::vtable(self).3)(
            ::windows::Abi::abi(self),
            appwindow.into_param().abi(),
            &<T as ::windows::Interface>::IID,
            ::windows::Abi::set_abi(&mut result__),
        )
        .and_some(result__)
    }
}
```

@kennykerr

**Comment by kennykerr** [comment]
That'll do it. I'll get that fixed up.

**Comment by kennykerr** [comment]
This is fixed as of 0.14.0.
